ui-react/packages/atlasmap/src/UI/useDoubleSelectDialog.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Modal,
+  Button,
+  FormSelect,
+  FormSelectOption,
+  Label,
+} from "@patternfly/react-core";
+import React, { ReactPortal, useCallback, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { CancelCallback } from "./useSingleInputDialog";
+
+interface ISelectValues1 {
+  isSelectable: boolean;
+  value: string;
+}
+
+export type ConfirmDoubleSelectCallback = (
+  value1: string,
+  value2: string,
+) => void;
+
+export interface IUseDoubleSelectDialogArgs {
+  title: string;
+  selectLabel1: string;
```

Comment:
For another day: I’m not a fan of this style of hooks where they are kinda generic but in reality they are specialized. We should simply have a confirm dialog that accepts a child element and provide as a child whatever logic we need - a form, a video, anything.

Also there is an issue with all the refs and states in use that make the performance of the dialog very bad. Try typing something eg. in the create constant dialog and you should see very heavy lag. That shouldn’t happen.

Reply:
This isn't currently used either.  Wanted to keep this around for teh user class select implementation.

Reply:
removed